### PR TITLE
feat(events): event closure job and admin status override

### LINKED ISSUE #237: feat(events): participation module MVP
## Problem Statement

A He4rt Developers organiza meetups presenciais, workshops e conferências (incluindo um meetup em breve e uma conferência de 2 dias no final do ano), mas não possui um sistema próprio para gerenciar o ciclo de participação dos membros. Hoje não há como:
- Controlar inscrições com capacidade limitada e lista de espera
- Verificar presença de quem realmente compareceu (check-in)
- Recompensar participação com XP no sistema de gamificação existente
- Aprovar candidatos a eventos com vagas limitadas (workshops)

## Solution

Construir o módulo de participação em eventos (`app-modules/events/`) com lifecycle completo: enrollment → check-in → attendance → XP reward. O sistema suporta 3 métodos de entrada (RSVP, RSVP+check-in, Application com formulário dinâmico), 3 métodos de check-in (manual, código numérico, QR code), waitlist com promoção FIFO, e integração com o módulo de gamificação via domain events.

## User Stories

1. As a participant, I want to RSVP to a meetup with one click, so that I can confirm my presence quickly
2. As a participant, I want to join a waitlist when an event is full, so that I can be promoted if someone cancels
3. As a participant, I want to be automatically promoted from the waitlist when a slot opens, so that I don't miss the opportunity
4. As a participant, I want to cancel my enrollment before the deadline, so that my slot is freed for others
5. As a participant, I want to check in at a meetup by typing a numeric code, so that my presence is verified
6. As a participant, I want to check in at a conference by showing my QR code badge, so that attendance is tracked per day
7. As a participant, I want to receive XP when I confirm, check in, and complete attendance, so that my participation is rewarded in the gamification system
8. As a participant, I want to fill an application form to apply for a workshop, so that the organizer can evaluate my candidacy
9. As a participant, I want to be notified when my application is approved or rejected, so that I know my status
10. As a participant, I want to check in on each day of a multi-day conference, so that my attendance per day is tracked
11. As a participant, I want to use the same QR code across all event days, so that I only need one badge
12. As a participant, I want to check in via Discord/Twitch bot by typing a command with the code, so that I can confirm presence in online events
13. As a participant, I want to see my enrollment status for each event, so that I know where I stand
14. As an organizer, I want to create events with configurable enrollment policies, so that each event has appropriate participation rules
15. As an organizer, I want to set capacity limits with optional waitlist, so that I can control attendance
16. As an organizer, I want to choose the enrollment method (RSVP, RSVP+check-in, Application), so that the entry flow matches the event type
17. As an organizer, I want to choose the check-in method (manual, numeric code, QR code), so that presence verification fits the event format
18. As an organizer, I want to define a dynamic application form (JSONB schema), so that I can collect relevant information per event
19. As an organizer, I want to approve or reject applications with a reason, so that participants understand the decision
20. As an organizer, I want to generate time-limited numeric codes bound to a specific date, so that check-in is controlled per day
21. As an organizer, I want to manually check in participants, so that I can handle edge cases (late arrivals, code issues)
22. As an organizer, I want to override a participant's status (e.g., no_show → attended), so that I can correct errors
23. As an organizer, I want to set a cancellation deadline in hours before the event, so that I have predictability on attendance
24. As an organizer, I want to configure XP rewards per event (RSVP, check-in, attendance), so that different events reward differently
25. As an organizer, I want to set attendance_requirement (all_days, any_day, minimum_days), so that multi-day events have clear completion criteria
26. As an organizer, I want to see all enrollments with their current status, so that I can manage the event
27. As an organizer, I want a full audit trail of every status transition, so that I can investigate disputes
28. As the system, I want to automatically mark checked_in → attended after the event ends, so that XP is awarded without manual intervention
29. As the system, I want to automatically mark confirmed → no_show after the event ends, so that non-attendees are identified
30. As the system, I want enrollment capacity checks to be atomic (lockForUpdate), so that overbooking is impossible under concurrency

### Architecture

- **Module**: `app-modules/events/` following existing modular architecture (`internachi/modular`)
- **Namespace**: `He4rt\Events\*`
- **Panels**: Admin panel = organizer, App panel = participant
- **Table prefix**: All tables prefixed with `events_` for clear identification in database

### State Machine

- Implemented as a PHP backed enum (`EnrollmentStatusEnum`) with `canTransitionTo()` method
- **NOT** using `spatie/laravel-model-states` — transitions validated in Action classes explicitly
- 8 states: `pending`, `confirmed`, `waitlisted`, `checked_in`, `attended`, `cancelled`, `rejected`, `no_show`
- Terminal states: `attended` (success), `cancelled`, `rejected`, `no_show`
- See ADR-0001

### Schema (7 tables)

**`events`** — Core event record:
- `id`, `tenant_id`, `event_type` (meetup/workshop/conference), `active`, `slug`, `title`, `description`, `location`, `starts_at`, `ends_at`, `timestamps`
- No denormalized counters — derived by query

**`events_enrollment_policies`** — 1:1 configuration per event:
- `event_id` (unique), `enrollment_method`, `check_in_method`, `capacity` (nullable = unlimited), `has_waitlist`, `attendance_requirement` (all_days/any_day/minimum_days), `cancellation_deadline_hours`, `xp_reward_rsvp`, `xp_reward_checkin`, `xp_reward_attendance`, `application_form_schema` (JSONB), `timestamps`

**`events_enrollments`** — Hub central (1 per user+event):
- `id`, `event_id`, `user_id`, `status` (enum), `waitlist_position` (nullable), `application_data` (JSONB), `rejection_reason`, `enrolled_at`, `confirmed_at`, `checked_in_at`, `attended_at`, `cancelled_at`, `is_public`, `timestamps`
- UNIQUE constraint on `(event_id, user_id)`

**`events_enrollment_transitions`** — Append-only audit trail:
- `id`, `enrollment_id`, `from_status`, `to_status`, `triggered_by` (system/user/admin), `actor_user_id` (nullable), `reason` (nullable), `created_at`
- Written explicitly by Actions, NOT by database triggers (ADR-0003)

**`events_check_ins`** — 1:N per enrollment (one per day):
- `id`, `enrollment_id`, `method` (manual/numeric_code/qr_code), `payload` (JSONB), `event_date`, `checked_in_at`, `timestamps`
- Multi-day events: one check-in record per day attended (ADR-0004)
- Event days derived from `events.starts_at`/`ends_at` date range — no separate days table

**`events_check_in_codes`** — Organizer-generated numeric codes:
- `id`, `event_id`, `code`, `event_date`, `valid_from`, `valid_until`, `max_uses` (nullable), `uses_count`, `timestamps`
- Bound to specific date — one code per day

**`events_qr_tokens`** — One per enrollment, reusable across days:
- `id`, `enrollment_id`, `token` (unique), `expires_at` (nullable), `timestamps`

### XP Integration

- Events module publishes domain events: `EnrollmentConfirmed`, `ParticipantCheckedIn`, `ParticipantAttended`
- Gamification module listens and calls `IncrementExperience` on Character
- XP amounts come from `events_enrollment_policies` fields, passed as payload in domain events
- No XP ledger in Events module — auditability via `events_enrollment_transitions`
- Idempotency responsibility on Gamification listener
- See ADR-0002

### Bot Integration

- Discord/Twitch bot dispatches a domain event (e.g., `CheckInRequested`) with user identifier and code
- Events module has a listener that validates the code and processes the check-in
- Bot is pure transport — all business logic in Events module
- Bot must listen for response event (`CheckInProcessed`) to reply to user in chat
- Consistent with existing Bot Discord → Moderation pattern in CONTEXT-MAP
- See ADR-0005

### Enrollment Flow — Concurrency

- Capacity check uses `DB::transaction` with `lockForUpdate()` on the enrollment policy
- Count confirmed/checked_in/attended enrollments → compare against capacity → insert or waitlist
- Waitlist promotion on cancellation: FIFO (ordered by `enrolled_at`), atomic update + notification dispatch

### Scheduled Job

- Runs after `event.ends_at` passes
- Batch transitions: `checked_in → attended` (if attendance_requirement met), `confirmed → no_show`
- Dispatches XP domain events for newly `attended` enrollments
- Single job handles both transitions per event

### Application Form

- Dynamic form schema stored as JSONB in `events_enrollment_policies.application_form_schema`
- Filament form builder for organizer to define questions
- Participant responses stored in `events_enrollments.application_data` as JSONB
- Organizer reviews in Admin panel with approve/reject actions

### Event Types

```php
enum EventTypeEnum: string {
    case Meetup = 'meetup';
    case Workshop = 'workshop';
    case Conference = 'conference';
}
```

## Testing Decisions

- **Good tests**: test external behavior through Actions and domain events, not internal implementation. Assert state transitions, database state, and dispatched events.
- **Test location**: `app-modules/events/tests/Feature/` and `app-modules/events/tests/Unit/`
- **Prior art**: existing tests in `app-modules/moderation/tests/` and `app-modules/bot-discord/tests/`
- **What to test**:
  - Each Action (EnrollUser, CancelEnrollment, CheckIn, ApproveApplication, RejectApplication, PromoteFromWaitlist) — happy path + edge cases
  - State machine transitions — valid and invalid
  - Concurrency: capacity enforcement under race conditions
  - Waitlist promotion FIFO ordering
  - Multi-day check-in with attendance_requirement validation
  - Scheduled job (attended + no_show marking)
  - Domain event dispatch (assert events are fired, don't test Gamification's listener here)
  - Cancellation deadline enforcement
  - Numeric code validation (expiry, max_uses, wrong date)
  - QR token validation (expiry, reuse across days)
- **Factories**: create factories for EventModel, EventEnrollmentPolicy, EventEnrollment, EventCheckIn, EventCheckInCode, EventQrToken
- **Use Pest** with `php artisan make:test --pest`
- **ActingAs**: all Filament panel tests require `$this->actingAs(User::factory()->create())`

## Out of Scope

- Networking between participants (profiles, connections, discovery)
- Referral / invite links with tracking
- Magic link check-in method
- Geolocation check-in method
- Sponsors association with events
- Timeline / activity feed display
- Call for Papers / Submissions (CFP)
- Agenda / schedule display
- Paid events / payment integration
- No-show penalties affecting future eligibility (data tracked, consequences deferred)
- Event days as explicit table (derived from date range)
- Denormalized counters on events table

## Further Notes

- The events module was previously gutted (all tables dropped in migration `2026_05_16_195205`). This PRD rebuilds the module from scratch with the new participation-focused architecture.
- The previous module had: events, events_talks, events_attendees, sponsors, events_sponsors, events_agenda, event_submission_speakers. The new module replaces this with a fundamentally different schema focused on enrollment lifecycle.
- Existing ADRs document key architectural decisions: see `app-modules/events/docs/adr/0001-0005`.
- `CONTEXT-MAP.md` at repo root has been updated to include Events and Gamification bounded contexts with dependency diagram.
- `app-modules/events/CONTEXT.md` contains the full domain glossary and state machine reference.
- Future phases (post-MVP): magic_link/geolocation check-in, CFP/submissions, agenda display, networking, referral, payment integration, no-show penalties.

### LINKED ISSUE #247: feat(events): event closure job (attended/no-show)
## Parent

#237

## What to build

Implement the scheduled job that runs after an event ends and automatically marks final states: `checked_in → attended` (success) and `confirmed → no_show` (absent).

**Job: `ProcessEventClosureJob`**
- Implements `ShouldQueue`, `ShouldBeUnique` (uniqueId = event_id)
- `$backoff = [1, 5, 10]` for exponential retry
- Implements `failed()` with logging
- Logic:
  1. Load event with policy
  2. Determine attendance_requirement from policy
  3. For each enrollment with status `checked_in`:
     - Count check-in records for this enrollment
     - Calculate event days from `starts_at` to `ends_at` (inclusive date range)
     - Apply requirement:
       - `all_days`: check-in count == total event days → `attended`
       - `any_day`: check-in count >= 1 → `attended` (always true if checked_in)
       - `minimum_days(N)`: check-in count >= N → `attended`
     - If requirement NOT met: → `no_show` (was present some days but not enough)
     - If met: transition to `attended`, set `attended_at`
  4. For each enrollment with status `confirmed` (never checked in): → `no_show`
  5. Record all transitions in audit trail (triggered_by=system)
  6. Dispatch `ParticipantAttended` domain event for each newly attended enrollment (implements `ShouldDispatchAfterCommit`)

**Scheduler:**
- Register job to run periodically (every 15 minutes or hourly)
- Only processes events where `ends_at < now()` AND has enrollments in non-terminal states
- Or: dispatch job once per event via `event:closure` artisan command

**Admin panel — Status override:**
- "Override Status" action on enrollment records
- Allows: `no_show → attended` (admin corrects mistake), `confirmed → checked_in` (late arrival that missed formal check-in)
- Requires reason text field
- Records transition with triggered_by=admin

## Acceptance criteria

- [ ] Job transitions `checked_in → attended` when attendance requirement is met
- [ ] Job transitions `confirmed → no_show` for participants who never checked in
- [ ] `attendance_requirement` is correctly evaluated:
  - `all_days`: requires check-in on every day of the event
  - `any_day`: any single check-in suffices
  - `minimum_days(N)`: requires N or more check-in days
- [ ] Job is idempotent — running twice produces same result (no duplicate transitions)
- [ ] `ShouldBeUnique` prevents duplicate job execution for same event
- [ ] `ParticipantAttended` domain event is dispatched for each newly attended enrollment
- [ ] Admin can override status (no_show → attended) with reason
- [ ] All transitions are recorded in audit trail with triggered_by=system
- [ ] Feature tests: all_days met, all_days not met, any_day, minimum_days, no_show marking, admin override, idempotency
- [ ] Pint passes

## Blocked by

- #244 (Check-in Manual — needs at least one check-in method to produce `checked_in` enrollments)

### PR DESCRIPTION
**Pull request by davicbtoliveira**
## Summary

Implements automated event closure and admin enrollment status override. After an event's `ends_at`, a scheduled job resolves all non-terminal enrollments to `attended` or `no_show`. Organizers can manually override `no_show → attended` or `confirmed → checked_in` via the admin panel with a required reason.

Closes #247

---

### Event closure pipeline

| File | Role |
|---|---|
| `src/Closure/Actions/CloseEventAction.php` | Core loop: per-enrollment `DB::transaction` with `lockForUpdate()`. Resolves checked-in enrollments via `EnrollmentPolicy::resolveAttendance()`, confirmed → `no_show`. Dispatches `ParticipantAttended` on success. Idempotent — skips terminal enrollments. |
| `src/Closure/Jobs/ProcessEventClosureJob.php` | `ShouldBeUnique` per `eventId` (30min lock), 4 tries, backoff `[1,5,10]`. Calls `CloseEventAction`. |
| `src/Console/Commands/ClosePendingEventsCommand.php` | `events:close-pending` — queries past events with non-terminal enrollments, dispatches one job per event. |
| `src/EventsServiceProvider.php` | Schedules command every 15 min with `withoutOverlapping()`. |
| `src/Closure/Events/ParticipantAttended.php` | Domain event consumed by gamification. `ShouldDispatchAfterCommit`. Payload: enrollmentId, eventId, userId, xpRewardOnAttended. |

### Admin status override

| File | Role |
|---|---|
| `src/Closure/Actions/OverrideEnrollmentStatusAction.php` | Separate action from `TransitionEnrollmentAction`. Allowlist: `no_show→attended`, `confirmed→checked_in`. Reason required. `triggered_by=admin` in audit trail. No domain event re-dispatch. |
| `src/Closure/DTOs/OverrideEnrollmentStatusDTO.php` | Input DTO: enrollment, toStatus, actorId, reason. |
| `src/Closure/Exceptions/OverrideEnrollmentStatusException.php` | `reasonRequired()`, `overrideNotAllowed(from, to)`. |
| `panel-admin/…/EnrollmentsRelationManager.php` | `overrideStatusAction()` — warning-colored action visible for no_show/confirmed. Modal with to_status select (dynamically populated from the Action's allowlist) + reason textarea. |

### EnrollmentPolicy improvements

| File | Role |
|---|---|
| `src/Enrollment/Models/EnrollmentPolicy.php` | `resolveAttendance(Enrollment): EnrollmentStatus` — counts check-in days against `attendance_requirement` (all_days, any_day, minimum_days). Returns `Attended` or `NoShow`. |
| `src/Enrollment/Enums/EnrollmentStatus.php` | Removed unused `getIcon()` + `HasIcon` interface — dead code. |

### EventForm (panel-admin)

| File | Role |
|---|---|
| `panel-admin/…/EventForm.php` | Attendance requirement fields: radio for requirement type, conditional minimum_days slider. Hidden for single-day events. |

### Architecture improvements

- Override allowlist now exposed as `OverrideEnrollmentStatusAction::allowedTargetsFor(EnrollmentStatus): list<EnrollmentStatus>` — single source of truth consumed by both validation and UI. `EnrollmentsRelationManager` no longer hardcodes allowed transitions.

---

## ADRs

- `app-modules/events/docs/adr/0007-event-closure-as-scheduled-job.md`
- `app-modules/events/docs/adr/0008-admin-status-override-as-separate-action.md`

---

## Test plan
- **CloseEventActionTest** (6 tests): all_days requirement, any_day, no-show, idempotency, multi-enrollment
- **ClosePendingEventsCommandTest** (4 tests): past/future event job dispatch, multiple events
- **ProcessEventClosureJobTest** (5 tests): closure, config (tries/backoff/uniqueFor via reflection), non-existent event, failure logging
- **OverrideEnrollmentStatusActionTest** (8 tests): allowed transitions, reason required, invalid pairs, `allowedTargetsFor()`
- **EnrollmentPolicyTest** (5 tests): all_days, any_day, minimum_days
- **EventResourceTest**: EventForm attendance requirement fields
- [x] PHPStan: 0 errors
- [x] Pint: clean

---

## Commits

| SHA | Message |
|---|---|
| `0ec3d42b` | `docs(events): adrs and context map` |
| `a2e0a803` | `feat(events): module structure (#249)` |
| `779f7598` | `feat(events): rsvp enrollment end-to-end (#275)` |
| `f631ea47` | `feat(events): waitlist and capacity enforcement (#294)` |
| `a51f4b7b` | `feat(events): numeric code check-in (#297)` |
| `8b3d3c41` | `feat(events): qr code check-in (#298)` |
| `c08f5042` | `pint + deps` |
| `30ef42b3` | `fix(events): resolve phpstan errors in check-in UI` |
| `d9a7f098` | `fix(events): correct tenant_id FK to UUID` |
| `022f471d` | `feat(enrollment): resolveAttendance()` |
| `da47e5c1` | `feat(panel_admin): override EnrollmentStatus` |
| `747dc31d` | `feat(events): schedule ClosePendingEventsCommand` |
| `4f6c111f` | `chore(exceptions): translations` |
| `d47fe97a` | `docs(adr): ADRs for issue #247` |
| `36ca23b4` | `feat(events): close-pending command` |
| `325d2840` | `feat(panel-admin): attendance requirements on EventForm` |
| `5d389511` | `feat(events): EnrollmentPolicyFactory fields` |
| `0fdf456e` | `feat(events): CloseEventAction` |
| `87e959a6` | `feat(events): OverrideEnrollmentStatusAction` |
| `821ef139` | `feat(events): OverrideEnrollmentStatusDTO` |
| `534a2840` | `feat(events): ProcessEventClosureJob` |
| `ed4b5b60` | `feat(events): OverrideEnrollment exceptions` |
| `f9dfa7fd` | `feat(events): ParticipantAttended event + tests` |
| `601efcca` | `chore(deps): package-lock.json` |
| `2e5f05fb` | `test(events): job config via reflection` |
| `4000f2ee` | `refactor(events): expose allowlist as queryable method` |
| `06703c7b` | `chore(events): remove unused getIcon()` |

### CODE REVIEW

File: app-modules/events/src/Closure/Actions/OverrideEnrollmentStatusAction.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            throw OverrideEnrollmentStatusException::reasonRequired();
+        }
+
+        return DB::transaction(function () use ($dto): EnrollmentTransition {
+            $enrollment = Enrollment::query()
+                ->whereKey($dto->enrollment->id)
+                ->lockForUpdate()
+                ->firstOrFail();
+
+            $fromStatus = $enrollment->status;
```
Comment (hefeus) [inline app-modules/events/src/Closure/Actions/OverrideEnrollmentStatusAction.php:45]: vale uma exceção mais específica para o caso de concorrência entre admins? Para no caso de dois admins atualizarem a mesma inscrição ao mesmo tempo? Por que isso tá vindo sem re-validação do DTO

File: app-modules/events/src/Closure/Actions/OverrideEnrollmentStatusAction.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    /**
+     * @throws OverrideEnrollmentStatusException
+     * @throws Throwable
+     */
+    public function handle(OverrideEnrollmentStatusDTO $dto): EnrollmentTransition
+    {
+        if (mb_trim($dto->reason) === '') {
+            throw OverrideEnrollmentStatusException::reasonRequired();
```
Comment (hefeus) [inline app-modules/events/src/Closure/Actions/OverrideEnrollmentStatusAction.php]: Essa validação poderia está dentro do DTO, que assim evitaria ter que passar ela toda vez que for usada novamente ou acabaram esquecendo essa validação.

File: app-modules/events/src/Closure/Actions/OverrideEnrollmentStatusAction.php [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        ['from' => EnrollmentStatus::Confirmed, 'to' => EnrollmentStatus::CheckedIn],
+    ];
+
+    /** @return list<EnrollmentStatus> */
+    public static function allowedTargetsFor(EnrollmentStatus $from): array
+    {
+        return array_values(array_map(
+            static fn (array $pair): EnrollmentStatus => $pair['to'],
+            array_filter(self::ALLOWED_OVERRIDES, static fn (array $pair): bool => $pair['from'] === $from),
+        ));
```
Comment (danielhe4rt) [inline app-modules/events/src/Closure/Actions/OverrideEnrollmentStatusAction.php]: Talvez usar as próprias collections do Laravel pra faciltiar a vida?

collect([])->values()->map(fn () => something)

File: app-modules/events/src/Closure/DTOs/OverrideEnrollmentStatusDTO.php [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+final readonly class OverrideEnrollmentStatusDTO
+{
+    public function __construct(
+        public Enrollment $enrollment,
+        public EnrollmentStatus $fromStatus,
+        public EnrollmentStatus $toStatus,
+        public string $actorId,
+        public string $reason,
+    ) {
+        if (mb_trim($this->reason) === '') {
```
Comment (danielhe4rt) [inline app-modules/events/src/Closure/DTOs/OverrideEnrollmentStatusDTO.php]:
```suggestion
        if (empty($this->reason)) {
```

File: app-modules/panel-admin/src/Filament/Resources/Events/RelationManagers/EnrollmentsRelationManager.php [hunk 5]
```diff
@@ -167,4 +172,42 @@ private function checkIn(Enrollment $enrollment, array $data): CheckIn
...
+                        actorId: (string) auth()->id(),
+                        reason: $data['reason'],
+                    ),
+                );
+
+                Notification::make()
+                    ->success()
+                    ->title('Enrollment status overridden.')
+                    ->send();
+            });
```
Comment (danielhe4rt) [inline app-modules/panel-admin/src/Filament/Resources/Events/RelationManagers/EnrollmentsRelationManager.php]: Extrair pra uma action separada.

File: app-modules/events/src/Closure/Actions/OverrideEnrollmentStatusAction.php (hunk 3 above)
Comment (davicbtoliveira) [reply app-modules/events/src/Closure/Actions/OverrideEnrollmentStatusAction.php]:
```php
    {
        return collect(self::ALLOWED_OVERRIDES)
            ->whereStrict('from', $from)
            ->pluck('to')
            ->values()
            ->all();
    }
```

File: app-modules/events/src/Closure/DTOs/OverrideEnrollmentStatusDTO.php (hunk 4 above)
Comment (davicbtoliveira) [reply app-modules/events/src/Closure/DTOs/OverrideEnrollmentStatusDTO.php]: Resolved using `blank()`, to validate if the reason is not only blank spaces

File: app-modules/panel-admin/src/Filament/Resources/Events/RelationManagers/EnrollmentsRelationManager.php (hunk 5 above)
Comment (davicbtoliveira) [reply app-modules/panel-admin/src/Filament/Resources/Events/RelationManagers/EnrollmentsRelationManager.php]: Done.

File: app-modules/events/src/Console/Commands/ClosePendingEventsCommand.php [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+{
+    public function handle(): int
+    {
+        $count = 0;
+
+        Event::query()
+            ->where('ends_at', '<', now())
+            ->whereHas('enrollments', function (Builder $query): void {
+                $query->whereIn('status', [EnrollmentStatus::Confirmed, EnrollmentStatus::CheckedIn]);
+            })
```
Comment (hefeus) [inline app-modules/events/src/Console/Commands/ClosePendingEventsCommand.php:27]: Adicionar um select para o id evitar de carregar todas as colunas para pegar somente o id, pois quando tiver muitos participantes levar menos tempo nisso seria ideal:

```
Event::query()
            ->where('ends_at', '<', now())
            ->whereHas('enrollments', function (Builder $query): void {
                $query->whereIn('status', [EnrollmentStatus::Confirmed, EnrollmentStatus::CheckedIn]);
            })
          ->select('id)  <- Adição desta linha
          ->each(function (Event $event) use (&$count): void {
                          dispatch(new ProcessEventClosureJob($event->id));
                          $count++;
                      });
          
```

**Review by danielhe4rt — APPROVED** [review]

**Review by YuriSouzaDev — APPROVED** [review]

**Review by gvieira18 — APPROVED** [review]

**Review by GabrielFVDev — APPROVED** [review]
